src/volatile-root/volatile-root.c

```diff
@@ -71,48 +74,74 @@ static int make_volatile(const char *root_path) {
         return r;
 }
 
-static int make_overlay(const char *root_path) {
-        _cleanup_free_ char *escaped_path = NULL;
+static int make_overlay(const char *root_path, int overlay_type) {
+        _cleanup_free_ char *escaped_path = NULL, *usr_path = NULL;
         bool tmpfs_mounted = false;
-        const char *options = NULL;
+        const char *options = NULL, *overlay_name = NULL;
+        _cleanup_free_ char *mount_path = NULL;
+        _cleanup_free_ char *overlay_name_dir = NULL;
+        _cleanup_free_ char *upper_dir = NULL;
+        _cleanup_free_ char *work_dir = NULL;
         int r;
 
         assert(root_path);
 
-        r = mkdir_p("/run/systemd/overlay-sysroot", 0700);
+        if (overlay_type == VOLATILE_OVERLAY) {
+                overlay_name = overlay_root_name;
+                mount_path = strdup(root_path);
+                if (!mount_path)
+                        return log_oom();
+        } else {
+                assert(overlay_type == VOLATILE_OVERLAY_USR);
+                overlay_name = overlay_usr_name;
+                mount_path = strjoin(root_path, "/usr");
+        }
+
+        overlay_name_dir = strjoin("/run/systemd/", overlay_name);
+        upper_dir = strjoin(overlay_name_dir, "/upper");
+        work_dir = strjoin(overlay_name_dir, "/work");
```

Comment:
And here too, three times.

In general, the existing code was using `strjoina`. This is fine, since we're operating on a limited number of strings with fixed size. I think the code will easier if this pattern is continued.